orc8r/cloud/go/mproto/test/mproto_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"magma/orc8r/cloud/go/mproto"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	iterationCount = 1000
+)
+
+var (
+	goldenFilepath = "../testdata/determinisic-digest.b64.golden"
+)
+
+// TestMarshalManyDeterministic checks if MarshalManyDeterministic truly enforces
+// deterministic encoding by comparing encoded protobuf messages containing the same data.
+//
+// Ref: https://gist.github.com/kchristidis/39c8b310fd9da43d515c4394c3cd9510
+func TestMarshalManyDeterministic(t *testing.T) {
+	// Encode basic proto messages (no compound fields).
+	protos1 := map[string]proto.Message{
+		"12345": &TestDataBasic{Key: "12345", Value: 10},
+		"23456": &TestDataBasic{Key: "23456", Value: 15},
+		"34567": &TestDataBasic{Key: "34567", Value: 20},
+	}
+	encodedCanon1, err1 := mproto.MarshalManyDeterministic(protos1)
+	assert.NoError(t, err1)
+
+	for i := 0; i < iterationCount; i++ {
+		encoded, err := mproto.MarshalManyDeterministic(protos1)
+		assert.NoError(t, err)
+		assert.Equal(t, encoded, encodedCanon1)
```

Comment:
Nit here and below: fn signature is "expected, actual", so we want to flip the args